app/models/piece.rb

```diff
@@ -11,14 +11,27 @@ def move_to!(new_row, new_col)
     check_if_castling(new_row, new_col) if type == "King"
     # Checking for Valid Move
     return unless valid_move?(new_row, new_col)
+    # Checking if the piece is moving into check
+    return unless !moving_into_check?(new_row, new_col)
```

Comment:
Yes, you're right! The reason we did it that way, is becuse moving_into_check? didn't return true before. After @jonathanpike's comment it does.